jupyter_client/manager.py

```diff
@@ -314,7 +320,7 @@ def format_kernel_cmd(self, extra_arguments: t.Optional[t.List[str]] = None) ->
 
         if self.kernel_spec:  # type:ignore[truthy-bool]
             ns["resource_dir"] = self.kernel_spec.resource_dir
-
+        assert self._launch_args is not None
```

Comment:
This helps mypy, the line below would fail anyway if it was None.

Reply:
| ok, I can't make it  t.Optional[Dict[str, Any]], as Dict is not subscriptable

You can use  `t.Optional["Dict[str, Any]"]` to get around the runtime issue.

Reply:
Ah ! True, let's try